tests/ui/let_and_return.rs

```diff
@@ -244,4 +252,15 @@ fn issue12801() {
     }
 }
 
+// Do not lint
```

Comment:
Should we change that to "Only lint in edition 2024 or later"?

Reply:
Good catch, thanks, comment removed in the second commit (and the error marker is explicit concerning which edition is concerned). I've let it in the first commit, as before separating the editions this is not linted.